core/src/main/scala/akka/kafka/internal/ConsumerStage.scala

```diff
@@ -84,6 +83,22 @@ private[kafka] object ConsumerStage {
         }
     }
 
+  def committableSourceWithMetadata[K, V](settings: ConsumerSettings[K, V],
+                                          subscription: Subscription,
+                                          _metadataFromRecord: ConsumerRecord[K, V] => String) =
+    new KafkaSourceStage[K, V, CommittableMessage[K, V]] {
+      override protected def logic(shape: SourceShape[CommittableMessage[K, V]]) =
+        new SingleSourceLogic[K, V, CommittableMessage[K, V]](shape, settings, subscription)
+        with CommittableMessageWithMetadataBuilder[K, V] {
+          override def metadataFromRecord(record: ConsumerRecord[K, V]): String = _metadataFromRecord(record)
```

Comment:
I believe this implementation may replace the existing `committableSource` version, if that one returns the empty String from `metadataFromRecord`.
Wouldn't that lead to fewer moving parts?